corehq/util/tests/test_validation.py

```diff
@@ -1,28 +1,33 @@
+from random import sample
+
 from corehq.util.validation import is_url_or_host_banned
 from django.test import TestCase
 
-
-def inclusive_range(start, stop):
-    return range(start, stop + 1)
+def sample_range(start, stop):
+    yield start
+    num_samples = min(8, stop - start)
+    for middle in sample(xrange(start + 1, stop), num_samples):
+        yield middle
+    yield stop
```

Comment:
:+1: just curious how much time this saved, i had thought it was all cpu bound so should have been very quick